framework/src/modules/token/module.ts

```diff
@@ -75,21 +88,235 @@ export class TokenModule extends BaseModule {
+			throw new Error(`Terminated escrow store must be sorted by chainID.`);
+		}
+		for (const terminatedChainID of genesisStore.terminatedEscrowSubstore) {
+			await terminatedEscrowStore.setWithSchema(
+				terminatedChainID,
+				{ escrowTerminated: true },
+				terminatedEscrowStoreSchema,
+			);
+		}
+
+		// verify result
+		// validateSupplyStoreEntries
+		const computedSupply = new dataStructures.BufferMap<bigint>();
+		const allUsers = await userStore.iterateWithSchema<UserStoreData>(
+			{
+				start: Buffer.alloc(26, 0),
+				end: Buffer.alloc(26, 255),
+			},
+			userStoreSchema,
+		);
+		for (const { key, value: user } of allUsers) {
+			const tokenID = key.slice(20);
+			const [chainID, localID] = splitTokenID(tokenID);
+			if (chainID.equals(CHAIN_ID_ALIAS_NATIVE)) {
+				const existingSupply = computedSupply.get(localID) ?? BigInt(0);
+				computedSupply.set(
+					localID,
+					existingSupply +
+						user.availableBalance +
+						user.lockedBalances.reduce((prev, current) => prev + current.amount, BigInt(0)),
+				);
+			}
+		}
+		const allEscrows = await escrowStore.iterateWithSchema<EscrowStoreData>(
+			{
+				start: Buffer.alloc(6, 0),
+				end: Buffer.alloc(6, 255),
+			},
+			escrowStoreSchema,
+		);
+		for (const { key, value } of allEscrows) {
+			const [, localID] = splitTokenID(key);
+			const existingSupply = computedSupply.get(localID) ?? BigInt(0);
+			computedSupply.set(localID, existingSupply + value.amount);
+		}
+		for (const [localID, supply] of computedSupply.entries()) {
+			if (!isUInt64(supply)) {
+				throw new Error(
+					`Total supply for LocalID: ${localID.toString('hex')} exceeds uint64 range.`,
+				);
+			}
+		}
+		const storedSupply = new dataStructures.BufferMap<bigint>();
+		const allSupplies = await supplyStore.iterateWithSchema<SupplyStoreData>(
+			{
+				start: Buffer.alloc(LOCAL_ID_LENGTH, 0),
+				end: Buffer.alloc(LOCAL_ID_LENGTH, 255),
+			},
+			supplyStoreSchema,
+		);
+		const maxLocalID = allSupplies[allSupplies.length - 1].key;
+		for (const { key, value } of allSupplies) {
+			storedSupply.set(key, value.totalSupply);
+		}
+
+		for (const [localID, supply] of computedSupply.entries()) {
+			const stored = storedSupply.get(localID);
+			if (!stored || stored !== supply) {
+				throw new Error('Stored total supply conflicts with computed supply.');
+			}
+		}
+		for (const [localID, supply] of storedSupply.entries()) {
+			if (!computedSupply.has(localID) && supply !== BigInt(0)) {
+				throw new Error('Stored total supply is non zero but cannot be computed.');
+			}
+		}
+
+		// validate next available ID
+		const {
+			nextAvailableLocalID,
+		} = await nextAvailableLocakIDStore.getWithSchema<AvailableLocalIDStoreData>(
+			EMPTY_BYTES,
+			availableLocalIDStoreSchema,
+		);
+		// If maxLocalID is larger than nextAvailableLocalID, it is invalid
+		if (maxLocalID.compare(nextAvailableLocalID) > 0) {
+			throw new Error('Max token local ID is higher than next availableLocalID');
```

Comment:
We reflect our variable naming in `availableLocalID` but don't reflect it in `Max token local ID`. `Max token local ID` is quite confusing while reading so I think either we should add hyphen to the words in it or reflect variable name in the error message to be more clear.

Reply:
I think Max token local ID was not matching as you mentioned, so I updated to Max local ID